### PR TITLE
[Bugfix] Update UI and handle new navigation

### PR DESCRIPTION
**Pull request by TackAdam**
### Description
1. Updates the UI with the compressed elements and formatting.
2. Add condition for new navigation to prevent double titles.
3. Update testing to handle chrome prop for new navigation check.

New navigation 
| Before | After |
| ------ | ----- |
| Before Image 1 | After Image 1 |
| Before Image 2 | After Image 2 |
| Before Image 3 | After Image 3 |
| Before Image 4 | After Image 4 |
| Before Image 5 | After Image 5 |

Old Navigation
| Before | After |
| ------ | ----- |
| Before Image 1 | After Image 1 |
| Before Image 2 | After Image 2 |
| Before Image 3 | After Image 3 |
| Before Image 4 | After Image 4 |
| Before Image 5 | After Image 5 |

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/report_definitions/create/create_report_definition.tsx [hunk 1]
```diff
@@ -301,12 +304,12 @@ export function CreateReport(props: { [x: string]: any; setBreadcrumbs?: any; ht
...
         <EuiTitle>
           <h1>
-            {i18n.translate('opensearch.reports.createReportDefinition.title', {
+            {!getNavGroupEnabled && i18n.translate('opensearch.reports.createReportDefinition.title', {
               defaultMessage: 'Create report definition',
             })}
           </h1>
         </EuiTitle>
-        <EuiSpacer />
+        <EuiSpacer size='s'/>
```
Comment (derek-ho) [inline public/components/report_definitions/create/create_report_definition.tsx:312]: QQ: Do we want to remove the extra spacing if the h1 is not present/empty? Let me know if this makes sense

File: public/components/main/report_definition_details/report_definition_details.tsx [hunk 2]
```diff
@@ -669,169 +668,167 @@ export function ReportDefinitionDetails(props: { match?: any; setBreadcrumbs?: a
...
+    <>
+      <EuiTitle size="l">
+        <h1>
+          {!getNavGroupEnabled && i18n.translate('opensearch.reports.reportDefinitionsDetails.title', {
+              defaultMessage: 'Report definition details',
+            }
+          )}
+        </h1>
+      </EuiTitle>
+      <EuiSpacer size="m" />
```
Comment (derek-ho) [inline public/components/main/report_definition_details/report_definition_details.tsx:680]: Same question as below

**Review by derek-ho — APPROVED** [review]
Mostly LGTM

File: public/components/report_definitions/create/create_report_definition.tsx (hunk 1 above)
Comment (TackAdam) [reply public/components/report_definitions/create/create_report_definition.tsx:312]: Good catch, fixed with the following 
{!getNavGroupEnabled && <EuiSpacer size='s' />}
Very minor look change for navigation enabled but makes sense!

File: public/components/main/report_definition_details/report_definition_details.tsx (hunk 2 above)
Comment (TackAdam) [reply public/components/main/report_definition_details/report_definition_details.tsx:680]: Fixed with {!getNavGroupEnabled && <EuiSpacer size='s' />}

**Comment by ps48** [comment]
Just one callout under the new nav, the table headers look as big as the title. May be we can reduce their size under the new nav?

Screenshot 2024-09-04 at 1 41 21 PM

**Review by ps48 — APPROVED** [review]
Nothing blocking thanks for taking this up

**Comment by TackAdam** [comment]
Adjusted spacing in old and new navigation. Added tittle to the old nav for it to match
New nav:
NewNav

Old nav:
OldNav

**Pull request merged**